syntax = "proto3";

package line_following;

service PositionServer {

    rpc registerClient (ClientInfo) returns (ServerInfo) {
    }

    rpc getPositions (ClientInfo) returns (stream Position) {
    }

}

message ClientInfo {
    string info = 1;
}

message ServerInfo {
    string info = 1;
}

message Position {
    int32 id = 1;
    bool in_focus = 2;
    int32 mid_offset = 3;
    int32 degrees = 4;
    int32 mid_line_cross = 5;
    int32 width = 6;
    int32 middle_inc = 7;
}


